R/graph.curve.params.R

```diff
@@ -49,183 +52,145 @@ graph.curve.params <- function( # nolint: object_name_linter
-    ggplot2::facet_wrap(
-      ~ .data$antigen_iso,
-      ncol = 2
-    ) +
+  plot1 <- ggplot2::ggplot() +
+    ggplot2::aes(x = .data$t, y = .data$res) +
+    ggplot2::facet_wrap(~ antigen_iso, ncol = 2) +
     ggplot2::theme_minimal() +
     ggplot2::theme(axis.line = ggplot2::element_line()) +
-    ggplot2::labs(x = "Days since fever onset",
-                  y = "ELISA units",
-                  col = if_else(show_all_curves, "MCMC chain", "")) +
+    ggplot2::labs(
+      x   = "Days since fever onset",
+      y   = "ELISA units",
+      col = if (show_all_curves) "MCMC chain" else NULL
+    ) +
     ggplot2::theme(legend.position = "bottom")
 
-  if (show_all_curves) {
-
-    range <-
-      serocourse_all |>
-      dplyr::summarize(
-        min = min(.data$res, 0.9),
-        max = max(.data$res, 2000)
-      )
 
-    group_vars <-
-      c("iter", "chain") |>
+  if (show_all_curves) {
+    group_vars <- c("iter", "chain") |>
       intersect(names(serocourse_all))
 
     if (length(group_vars) > 1) {
       serocourse_all <-
         serocourse_all |>
-        mutate(
-          iter = interaction(across(all_of(group_vars)))
+        dplyr::mutate(
+          iter = interaction(
+            dplyr::across(dplyr::all_of(group_vars))
+          )
         )
+
       plot1 <-
         plot1 +
-        geom_line(
+        ggplot2::geom_line(
           data = serocourse_all,
           alpha = alpha_samples,
           aes(
-            color = .data$chain |> factor(),
+            color = factor(.data$chain),
             group = .data$iter
           )
-        ) +
-        ggplot2::expand_limits(y = range)
+        )
     } else {
-
       plot1 <-
         plot1 +
-        geom_line(data = serocourse_all,
-                  alpha = alpha_samples,
-                  aes(group = .data$iter)) +
-        ggplot2::expand_limits(y = range)
-
+        ggplot2::geom_line(
+          data = serocourse_all,
+          alpha = alpha_samples,
+          aes(group = .data$iter)
+        )
     }
-
+    plot1 <-
+      plot1 + ggplot2::expand_limits(y = unlist(range))
   }
 
-  plot1 <-
-    plot1 +
+  plot1 <- plot1 +
     ggplot2::scale_y_log10(
       limits = unlist(range),
       labels = scales::label_comma(),
       minor_breaks = NULL
     )
 
-  if (show_quantiles) {
-    plot1 <-
-      plot1 +
-      ggplot2::geom_line(
-        ggplot2::aes(col = "median"),
-        data = serocourse_sum |> filter(.data$quantile == "med"),
-        linewidth = 1
-      ) +
+  if (!is.null(quantiles)) {
+    plot1 <- plot1 +
       ggplot2::geom_line(
-        ggplot2::aes(col = "10% quantile"),
-        data = serocourse_sum |> filter(quantile == "p10"),
-        linewidth = .5
-      ) +
-      ggplot2::geom_line(
-        ggplot2::aes(col = "90% quantile"),
-        data = serocourse_sum |> filter(quantile == "p90"),
-        linewidth = .5
+        data = serocourse_sum,
+        aes(
+          color = paste0("q", .data$quantile),
```

Comment:
I think the legend was better before (see [test](https://github.com/UCD-SERG/serocalculator/pull/434/files#diff-f6a25b123cb1b274522895c4945490f0c490ba6bb985e532a617f9c649a4323b)); how about:

```suggestion
          color = paste0(.data$quantile * 100, "% quantile"),
```